src-ui/src/app/components/app-frame/app-frame.component.ts

```diff
@@ -92,6 +94,25 @@ export class AppFrameComponent
       this.checkForUpdates()
     }
     this.tasksService.reload()
+
+    this.messagesService
+      .get()
```

Comment:
Can you explain exactly why this is needed (I get what it does I mean what situation it’s needed)? This service now runs one time only, which seems like a bit of a waste

Reply:
In some cases allauth is just indicating an error via a django message. For example if the account cannot be linked: https://github.com/pennersr/django-allauth/blob/4c3b84a058b4bf3f9ea8fac72e053e90039bbebc/allauth/socialaccount/helpers.py#L197

I just went for the minimal implementation to get access to the messages. And in this specific case for authentication I think it always enough to just check on page load of the frontend for new messages. I guess anything else would need to be based on polling for messages from django. I'm not aware of any hook we could use to have django automatically push new messages.

Or if you have another idea for the error handling I'm also open to that.

Reply:
So if I understand correctly, if you want to use these new buttons under profile to login and theres an error:
1. User clicks button from frontend --> opens login URL in new tab
2. Proceeds via login flow in whatever app
3. User is redirected back to paperless in the same browser window, at which point app loads & checks for Django messages that might be present.

In this case, the login button probably shouldn't have `_blank` since you end up with two windows of the app, I'll change that.

For the error messages, I guess we dont have much control over what happens once the user enters the allauth flow process. Maybe we can embed the message in the page or something simple like that, rather than having a whole new API endpoint, frontend service, etc. that only gets used once. 

But yea, I need to think about it / play with it.

Reply:
I just found that all calls to `add_message` from `allauth` are going through the adapter which we subclass anyway already: https://github.com/pennersr/django-allauth/blob/4c3b84a058b4bf3f9ea8fac72e053e90039bbebc/allauth/account/adapter.py#L367

So we could hook into that to transport the messages more like push notifications to the frontend. Could the websocket be reused for this?

Reply:
Yea I thought about the websocket. It would need to be adapted quite a bit as at the moment it's only expecting consumer messages, which also seems sorta not worth it because this basically only happens once, right? But the other question is whether the timing would work, i.e. when the backend calls `add_message` the user is still in the process of being redirected in which case by the time they load the frontend UI that message wont be received (I think). Django queues them until someone loads a page which is why it works there, your current implementation effectively does the same i.e. just pulls from the queue once the UI loads.

Will keep brainstorming, I havent looked at it yet much myself. Its not the end of the world if it stays like this.

Reply:
Ok, I just pushed a change that should sort this, I think. Any pending messages are just embedded in the html and the frontend picks them up on start, once. I tested this out a bit, of course let me know if you notice any issues.

<img width="402" alt="Screenshot 2024-01-05 at 10 34 33 AM" src="https://github.com/paperless-ngx/paperless-ngx/assets/4887959/f9d63e65-edf4-4385-bcdf-dfa69b4259cd">

Reply:
Looks good to me